src/gwt/panmirror/src/editor/src/optional/ace/ace.ts

```diff
@@ -711,6 +716,15 @@ export class AceNodeView implements NodeView {
         this.scrollRow = -1;
       }),
     );
+
+    // Keep track of mouse state so we can avoid e.g., autoscrolling while the
+    // mouse is down
+    this.dom.addEventListener("mousedown", (evt) => {
+      this.mouseDown = true;
+    });
+    this.dom.addEventListener("mouseup", (evt) => {
```

Comment:
What happens if you initiate a click / mousedown within the visual editor surface, but then later mouseup somewhere else? Does this event handler still see the mouseup event?

Reply:
It's a weird thing to do (you'd be generating a selection with one foot in code and the other in prose), but it would indeed leave this flag stuck if you did. I'll add a `mouseleave` handler to guard against this.